remapping_commands.sh

```diff
@@ -0,0 +1,195 @@
+#bed file
+#Input: 
+#[chr]	[pos]	[rsID]	[ALT]	[QUAL]	[FILT]	[INFO]
+#Output:
+#[chr]	[pos-1]	[pos]	[rsID]	[ALT]	[QUAL]	[FILT]	[INFO]
+awk '{print $1"\t"$2-1"\t"$2"\t"$3"\t"$4"\t"$5"\t"$6"\t"$7}' variants_remapped.vcf > variants_remapped.bed
+
+#getfasta to get the REF genome alleles and then extract the genome alleles without the positions
+bedtools getfasta -tab -fi ../"$newgenome" -bed variants_remapped.bed | awk '{print $2}'> genome_alleles.txt
+#make all alleles caps (sometimes there are lower-case bases)
+tr a-z A-Z < genome_alleles.txt > genome_alleles.fixed.txt
+
+#paste all the information into the correct columns
+cut -f1-3 variants_remapped.vcf > temp_first_3_columns.txt
+cut -f4-7 variants_remapped.vcf > temp_last_4_columns.txt
+paste temp_first_3_columns.txt genome_alleles.fixed.txt temp_last_4_columns.txt > var_pre_final.vcf
+
+#header:
+#create list of contigs/chromosomes to be added to the header
+awk '{print $1}' var_pre_final.vcf | sort | uniq > contig_names.txt
+while read CHR; do echo "##contig=<ID=$CHR>"; done < contig_names.txt > contigs.txt
+
+#add the reference assembly
+echo "##reference=$newgenomeaccession" >> contigs.txt
+
+#copy everything that isn't #CHROM (the column titles), ##contig or ##reference from the old header to a temp
+awk '($1 !~ /^##contig/ && $1 !~ /^##reference/ && $1 !~ /^#CHROM/) {print $0}' vcf_header.txt > temp_header.txt
+
+#add the two headers together and add the column names
+cat temp_header.txt contigs.txt > final_header.txt
+echo -e "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO" >> final_header.txt
+
+#add header to the vcf file:
+cat final_header.txt | cat - var_pre_final.vcf > temp && mv temp vcf_out_with_header.vcf
+#also add it to the ref alleles file so that the line numbers match:
+cat final_header.txt | cat - old_ref_alleles.txt > temp && mv temp old_ref_alleles_with_header.txt
+
+#test each variant to see if REF = ALT, if so, replace the current REF with the corresponding old REF (this is to deal 
+#with variants like G > G)
+.././replace_refs.py -i vcf_out_with_header.vcf -r old_ref_alleles_with_header.txt -o pre_final_vcf.vcf
+
+#run bcftools norm to swap the REF and ALT alleles if the REF doesn't match the new assembly
+bgzip pre_final_vcf.vcf
+bcftools norm -c ws -f ../"$newgenome" -N pre_final_vcf.vcf.gz -o ../"$outfile" -O v
+
+cd ../
+bgzip "$outfile"
+bcftools stats "$outfile".gz > stats.txt
+gzip -d "$outfile".gz
+
+cd "$TMPDIR"
+echo "-----------------------------------Results-----------------------------------"
+echo "Total number of input variants:"
+grep "^[^#]" ../"$vcffile" | wc -l
+echo "Total number of variants processed (after filters):"
+total=$(grep "^[^>]" variant_reads.out.fa | wc -l)
+echo $total
+echo "Number of remapped variants:"
+remapped=$(grep "^[^#]" ../"$outfile" | wc -l)
+echo $remapped
+echo "Percentage of remapped variants:"
+percentage=$(bc <<< "scale=3;($remapped/$total)*100")
+echo $percentage"%"
+
+cd ../
```

Comment:
In the future we might add the possibility for a user to specify the location of the desired temporary directory. In this case, `cd ../` will not necessarily return us to where we were. It's more reliable to use `cd -` (go back to the previous directory)